### PR TITLE
PXE boot modifications and fixes.

### PR DESCRIPTION
**Pull request by sbluhm**
## What does this PR change?

1. The tftp server requires read access to the boot images. Those are hard linked from the Uyuni repo location. On RHEL, the tftp server runs as user root. Thus `a+r`should be set to allow access to clients.

2.  Explicitly download the treeinfo file if it is found. `.treeinfo` might be skipped otherwise as it might be hidden by the server. Missing .treeinfo file might prevent a kvm kickstart.

3. Cobbler syncs the boot loaders to the tftpboot folder for the tftp server. The loaders folder is empty on RHEL and contains selected soft links to the syslinux folder on SUSE. The cobbler user has the choice to run `cobbler get-loaders` to download the boot loaders or to use the already available syslinux files.
This modification configures the direct usage of the syslinux folder to avoid (manual) downloads on a production setup. In addition it is expected to make the workaround bnc#668908 unnecessary.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: No functional changes.

- [X] **DONE**

## Test coverage
- No tests: unknown how to. Tested changes on CentOS 8.

- [X] **DONE**

## Links

Spacewalk reference: https://github.com/spacewalkproject/spacewalk/pull/702

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: backend/satellite_tools/reposync.py [hunk 1]
```diff
@@ -684,9 +684,9 @@ def sync(self, update_repodata=True):
         fileutils.createPath(os.path.join(CFG.MOUNT_POINT, 'rhn'))  # if the directory exists update ownership only
         for root, dirs, files in os.walk(os.path.join(CFG.MOUNT_POINT, 'rhn')):
             for d in dirs:
-                fileutils.setPermsPath(os.path.join(root, d), group=APACHE_GROUP)
+                fileutils.setPermsPath(os.path.join(root, d), group=APACHE_GROUP, chmod=int('0755', 8))
             for f in files:
-                fileutils.setPermsPath(os.path.join(root, f), group=APACHE_GROUP)
+                fileutils.setPermsPath(os.path.join(root, f), group=APACHE_GROUP, chmod=int('0754', 8))
```
Comment (mcalmer) [inline backend/satellite_tools/reposync.py:689]: `0754` for "files" is very unusual. Or did you want to set it to `0644` ?

Comment (sbluhm) [reply backend/satellite_tools/reposync.py:689]: I have actually just taken the existing permissions and added the `read` rights for `other`. `644` will also work for the purpose of this PR. I am unsure of any other impact of removing `execute` rights for `owner`/`group`.

Comment (mcalmer) [reply backend/satellite_tools/reposync.py:689]: I think we should try with `0644` for the files. For file transfer we should not need execute permissions.

File: backend/satellite_tools/reposync.py [hunk 2]
```diff
@@ -684,9 +684,9 @@ def sync(self, update_repodata=True):
         fileutils.createPath(os.path.join(CFG.MOUNT_POINT, 'rhn'))  # if the directory exists update ownership only
         for root, dirs, files in os.walk(os.path.join(CFG.MOUNT_POINT, 'rhn')):
             for d in dirs:
-                fileutils.setPermsPath(os.path.join(root, d), group=APACHE_GROUP)
+                fileutils.setPermsPath(os.path.join(root, d), group=APACHE_GROUP, chmod=int('0644', 8))
```
Comment (mcalmer) [inline backend/satellite_tools/reposync.py:687]: Oh, here we iterate over directories. They need of cause 755 permissions. Below we iterate over files and they need 644.

**Review by mcalmer — CHANGES_REQUESTED** [review]

File: backend/satellite_tools/reposync.py (hunk 2 above)
Comment (sbluhm) [reply backend/satellite_tools/reposync.py]: I saw that, thought the same and then just continued...

**Review by mcalmer — APPROVED** [review]

**Review by brejoc — APPROVED** [review]
I've also asked Thomas, since I don't know much about syslinux and this looks good to both of us.

**Comment by mcalmer** [comment]
@sbluhm would you rebase this PR on top of current master and solve the conflicts in the changes files please?
Then I would just merge.

**Pull request merged**